espree.js

```diff
@@ -786,14 +824,32 @@ function scanStringLiteral() {
             if (!ch || !syntax.isLineTerminator(ch.charCodeAt(0))) {
                 switch (ch) {
                 case "u":
+                    if (source[index] === "{") {
+                        if (extra.ecmaFeatures.unicodeCodePointEscapes) {
+                            ++index;
+                            str += scanUnicodeCodePointEscape();
+                        } else {
+                            throwError({}, Messages.UnexpectedToken, "ILLEGAL");
+                        }
+                    } else {
+                        restore = index;
+                        unescaped = scanHexEscape("u");
```

Comment:
I'd thought about doing that originally, but decided it would be too clever / easy to misunderstand. I'll try it with some documentation.